pkg/monitor/cluster/cluster.go

```diff
@@ -86,20 +89,31 @@ func NewMonitor(ctx context.Context, log *logrus.Entry, restConfig *rest.Config,
 		return nil, err
 	}
 
+	var hiveclientset *hiveclient.Clientset
```

Comment:
This should probably be an interface. I also wondering if it is the casue of the issue with `nil` checks.

```suggestion
	var hiveclientset hiveclient.Interface
```

Reply:
Is an interface now.
Could go back to the initial version of the nil check.